scripts/mem_usage.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2014-2017, Linaro Limited
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+
+def get_args():
+
+	parser = argparse.ArgumentParser(
+		description='Shows the memory layout of the TEE ELF file. '
+		'Size is provided for each section.')
+	parser.add_argument('tee_elf', help='The OP-TEE ELF file (tee.elf)')
+	return parser.parse_args()
+
+def printf(format, *args):
+	sys.stdout.write(format % args)
+
+def print_sect(name, addr, size, round_up, print_num_pages):
+	if size == 0:
+		size_kib = 0;
+		num_pages = 0;
+	else:
+		if round_up:
+			size_kib = (size - 1) / 1024 + 1
+		else:
+			size_kib = size / 1024
+		num_pages = (size - 1) / 4096 + 1
+
+	printf('%-16s %.8X - %.8X size %.8X %3d KiB', name, addr, addr + size,
+	       size, size_kib)
+	if print_num_pages:
+		printf(' %d pages', num_pages)
+	printf('\n')
+
+def readelf_cmd():
+	return os.getenv('CROSS_COMPILE', '') + 'readelf'
+
+def main():
+
+	in_shdr = False
+	sects = []
+
+	args = get_args()
+
+	readelf = subprocess.Popen([readelf_cmd(), '-a', '-W', args.tee_elf],
+				   stdout = subprocess.PIPE)
+	for line in iter(readelf.stdout.readline, ''):
+		if 'Section Headers:' in line:
+			in_shdr = True
+			continue
+		if 'Key to Flags:' in line:
+			in_shdr = False
+			continue
+		if in_shdr:
+			words = line.split()
+			if words[0] == '[':
```

Comment:
`if words.startswith('[')` is more pythonic

Reply:
No, words is a list, not a string

Reply:
Ah, yes. Missed that. Then it is okay.